src/components/common/PageForm.tsx

```diff
@@ -17,16 +18,24 @@ export interface PageFormProps {
 }
 
 export default function PageForm(props: PageFormProps): JSX.Element {
-  const { children, className, hideEdit, ...bottomBarProps } = props;
+  const { children, className, hideEdit, onSave, ...bottomBarProps } = props;
   const theme = useTheme();
   const { isMobile } = useDeviceInfo();
+  const [processing, setProcessing] = useState(false);
+
+  const handleSave = async () => {
+    setProcessing(true);
+    await onSave(); // <-- we want all onSaves to be async
```

Comment:
thanks..
in most cases onSave will redirect user to some other location and this component will unmount.. react might complain on the setProcessing(false) saying component was already unmounted.. 
we could live with that? unless there's some way to detect that..

Reply:
Hmm that's true, I think we can live with that for now.